### PR TITLE
[T.F 2.0 API Docs] Modifying documentation of set_optimizer in config.py

### PR DESCRIPTION
**Pull request by Harsh188**
PR for issue #42450 

Making the modification to the existing functions:
- `set_optimizer_experimental_options`

### CODE REVIEW

**Comment by Harsh188** [comment]
@googlebot I fixed it.

**Comment by MarkDaoust** [comment]
@Harsh188, I see you have signed the CLA. 

But the bot doesn't just check the user submitting the PR, it checks the email address in the commits.

Can you probably fix it with a `git commit --amend --author="John Doe <john@doe.org>"` and a `git push -f`.

**Comment by gbaned** [comment]
@Harsh188 Can you please make sure to use same GitHub username and email-id associated with it

**Comment by Harsh188** [comment]
@gbaned @MarkDaoust I'm using the same GitHub username and email associated with my GitHub account. Looks like this issue came up because I reinstalled my macOS and my last name was listed as Mohan instead of MohanKumar hopefully this change fixes my issues with other PRs.

**Comment by Harsh188** [comment]
Looks like after reinstalling macOS on my computer I had my last name listed as _Mohan_ instead of _MohanKumar_ :P

File: tensorflow/python/framework/config.py [hunk 1]
```diff
@@ -195,39 +195,59 @@ def get_optimizer_experimental_options():
 def set_optimizer_experimental_options(options):
   """Set experimental optimizer options.
 
-  Note that optimizations are only applied in graph mode, (within tf.function).
-  In addition, as these are experimental options, the list is subject to change.
+  **Note:** Optimizations are only applied in graph mode,
+  (within `tf.function`).
+
+  By default all of the optimizers are set to False.
```
Comment (jaingaurav) [inline tensorflow/python/framework/config.py:201]: This is not correct, the defaults values are a bit hard to determine but they are set here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/meta_optimizer.cc#L228. I think you need to rephrase this to note to something of the effect that in order to use a non-default option or ensure a certain optimization is used/skipped.

**Review by jaingaurav — CHANGES_REQUESTED** [review]

File: tensorflow/python/framework/config.py (hunk 1 above)
Comment (Harsh188) [reply tensorflow/python/framework/config.py:201]: Ah, thank you!

**Comment by Harsh188** [comment]
Hi @jaingaurav, can you help me sort this out?

It looks like all of the default optimizers are being pushed onto the `optimizer` vector in [`meta_optimizers.cc`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/meta_optimizer.cc#L228). This leads me to believe that all of the default optimizers are active. 

This seems a bit confusing because when I run 
```python
>>> tf.config.get_optimizer_experimental_options()
```
the output I get is `{'disable_model_pruning': False, 'disable_meta_optimizer': False}`. 

It doesn't list out all of the optimizers that are initialized to `True`. This is what initially lead me to believe that they were all set to `False`. 

This can be observed in `context.py` where [`_optimizer_experimental_options`](https://github.com/tensorflow/tensorflow/blob/8b7d5b4842d9a6107ca0df319c488d4ac27d8f20/tensorflow/python/eager/context.py#L451) is initially set to an empty dictionary. This results in the state of the default optimizers not being added to the dictionary when [`get_optimizer_experimental_options`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/eager/context.py#L1533) is called.

**Comment by jaingaurav** [comment]
@Harsh188: You are right, this is a problem. The API basically allows setting a value, but doesn't tell you the default that are already set. This is very confusing and largely a problem with the interplay between the python configuration interface and the C++ state. What we need is an API to the C++ layer that tells what optimizers are set once the context is initialized.

**Comment by Harsh188** [comment]
I'm gonna open up another PR to fix that issue and then get back to this since I'm still unsure about what default optimizers are turned on.
